### ISSUE TITLE
findGitDirectory can return an invalid git directory

### ISSUE THREAD

**Issue by icecream17** [atom/atom#22577]
### Prerequisites

* [ ] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: <https://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode>
    * Followed all applicable steps in the debugging guide: <https://flight-manual.atom.io/hacking-atom/sections/debugging/>
    * Checked the FAQs on the message board for common solutions: <https://discuss.atom.io/c/faq>
    * Checked that your issue isn't already filed: <https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom>
    * Checked that there is not already an Atom package that provides the described functionality: <https://atom.io/packages>

### Description

This is part of the pr at #22578.
This is the only section filled out, since #22578 is already adequate for explanation.

Basically the code doesn't await the promise returned by `isValidGitDirectory`, thereby not actually checking if the git directory is valid.

### Steps to Reproduce

1. 
2. 
3. 

**Expected behavior:**

**Actual behavior:**

**Reproduces how often:**
